recipes/soplex/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import collect_libs, copy, get, replace_in_file
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.scm import Version
+from os.path import join
+
+required_conan_version = ">=1.53.0"
+
+
+class SoPlexConan(ConanFile):
+    name = "soplex"
+    description = "SoPlex linear programming solver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://soplex.zib.de"
+    topics = ("simplex", "solver", "linear", "programming")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "BOOST": [True, False],
+        "GMP": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "BOOST": False,
+        "GMP": False
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def _determine_lib_name(self):
+        if self.options.shared:
+            return "soplexshared"
+        elif self.options.get_safe("fPIC"):
+            return "soplex-pic"
+        else:
+            return "soplex"
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        replace_in_file(self, "CMakeLists.txt", "GMP_INCLUDE_DIRS", "gmp_INCLUDE_DIRS")
+        replace_in_file(self, "CMakeLists.txt", "GMP_LIBRARIES", "gmp_LIBRARIES")
```

Comment:
It is probably better to change the generated files then to patch the should code https://github.com/conan-io/conan-center-index/blob/34a9e9b0ae4c7cef1d6fae21c6bdcfefbcfde246/recipes/qt/6.x.x/conanfile.py#L464 for example should work in this case

Reply:
I don't know how to set include and library paths with `set_property`. Which one is it? https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#properties

Reply:
Ohh, target name should == variable name, it just takes the same value

Reply:
I was not able to make it work with `set_property`. but with `cache_variables`